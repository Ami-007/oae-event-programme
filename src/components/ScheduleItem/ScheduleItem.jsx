import React, { useState, useEffect } from "react";
import "./ScheduleItem.scss";
import chevronDown from "../../assets/images/chevron-down.png";
import chevronUp from "../../assets/images/chevron-up.png";
import Description from "../Description/Description";

const ScheduleItem = (props) => {
  const { header, subHeader, description, alternateDescription } = props;
  const [showText, setShowText] = useState(false);
  const [Img, setImg] = useState(chevronDown);

  useEffect(() => {
    if (description === null) {
      setImg("");
    }
  }, [description]);

  const handleClick = () => {
    setShowText(!showText);
    if (showText) {
      setImg(chevronDown);
    } else {
      setImg(chevronUp);
    }
  };

  return (
    <div className="schedule-item">
      <div className="schedule-item__top">
        <span className="schedule-item__headers">
          <h3 className="schedule-item__header">{header}</h3>
          <p className="schedule-item__subheader">{subHeader}</p>
        </span>
        <span className="schedule-item__button-container">
          <img
            role="button"
            className="schedule-item__button"
            src={Img}
            alt=""
            onClick={handleClick}
          ></img>
        </span>
      </div>
      {showText && (
        <Description text={description} altText={alternateDescription} />
      )}
      <hr className="schedule-item__hr"></hr>
    </div>
  );
};

export default ScheduleItem;
