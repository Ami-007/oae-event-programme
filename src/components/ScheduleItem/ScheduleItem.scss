@import "../../assets/sass/typography";
@import "../../assets/sass/variables";

.schedule-item {
  display: flex;
  flex-direction: column;
  align-items: center;
  line-height: 1.5rem;
  font-family: $modern-content;
  width: 100%;

  &__top-div {
    font-family: $modern-header;
    display: flex;
    align-items: center;
    justify-content: space-between;
    align-items: baseline;
    width: 100%;
    padding: 0.5rem;
  }
  &__top-button {
    border: none;
    padding: 0;
    overflow: visible;
    background: transparent;
    font-family: $modern-header;
    text-align: start;
    display: flex;
    justify-content: space-between;
    align-items: baseline;
    width: 100%;
    color: inherit;
    font: inherit;
    line-height: normal;
  }
  &__subheader {
    color: rgb(156, 156, 156);
  }
  &__content {
    word-wrap: break-word;
    width: 100%;
  }
  &__image-conatiner {
    display: flex;
  }
  &__hr {
    width: 100%;
    color: lightgray;
    margin: 0 0.5rem 0 0.5rem;
  }
}
