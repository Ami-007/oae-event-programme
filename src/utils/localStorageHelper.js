export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
