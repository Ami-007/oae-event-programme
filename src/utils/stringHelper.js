export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
