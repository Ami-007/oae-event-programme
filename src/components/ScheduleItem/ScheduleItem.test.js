import { screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils"
import userEvent from "@testing-library/user-event";
import ScheduleItem from "./ScheduleItem"

test("Renders App component with default page", () => {
  const { container } = customRender(<ScheduleItem />);
  expect(container).toMatchSnapshot();
});

it("description should not render on load", () => {
  customRender(<ScheduleItem header="header" subHeader="subheader" description="lorem ipsum whatever"/>);
  const description = screen.queryByText("lorem ipsum whatever");
  expect(description).not.toBeInTheDocument();
})
it("header should render and description should render on button click", () => {
  customRender(<ScheduleItem header="header" subHeader="subheader" description="lorem ipsum whatever"/>);
  const header = screen.queryByText("header");
  expect(header).toBeInTheDocument()
  const button = screen.getByRole("button");
  userEvent.click(button)
  const description = screen.queryAllByText("lorem ipsum whatever")  
  expect(description[0]).toBeVisible()
})
it("should render the buuton and change the picture on click of the button", () => {
  customRender(<ScheduleItem header="header" subHeader="subheader" description="lorem ipsum whatever"/>); 
  const button = screen.getByRole("button");
  expect(button).toBeInTheDocument();
  expect(button).toHaveAttribute("src", "chevron-down.png");
  userEvent.click(button)
  expect(button).toHaveAttribute("src", "chevron-up.png");
})
it("should not render the button when there is no description", () => { 
  customRender(<ScheduleItem header="header" subHeader="subheader" />);
  const button = screen.getByRole("button");
  expect(button).toHaveAttribute("src", "");
})



