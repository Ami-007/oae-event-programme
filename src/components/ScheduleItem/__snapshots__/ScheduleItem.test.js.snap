// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App component with default page 1`] = `
<div>
  <div
    class="schedule-item"
  >
    <button
      class="schedule-item__top-button"
    >
      <span
        class="schedule-item__headers"
      >
        <h3
          class="schedule-item__header"
        >
          header
        </h3>
        <p
          class="schedule-item__subheader"
        >
          subheader
        </p>
      </span>
      <span
        class="schedule-item__icon-container"
      >
        <img
          alt=""
          class="schedule-item__icon"
          src="chevron-down.png"
        />
      </span>
    </button>
    <hr
      class="schedule-item__hr"
    />
  </div>
</div>
`;
